src/test/java/com/github/martapietka/ppm2pbmSpring/ImageServiceTest.java

```diff
@@ -0,0 +1,54 @@
+package com.github.martapietka.ppm2pbmSpring;
+
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.io.InputStream;
+
+@SpringBootTest
+class ImageServiceTest {
+
+    @Autowired
+    private ImageService imageService;
+
+    @Test
+    public void testGetConvertedInputStream() throws Exception {
+
+        String imageName = "3x3.ppm";
+
+        String expectedOutputString = """
+                P1
+                3 3
+                1 0 1
+                0 1 0
+                0 0 1
+                """;
+
+        try (InputStream inputStream = getClass().getResourceAsStream("/" + imageName)) {
```

Comment:
Zastanów się, czy przypadkiem nie masz już gdzieś podobnego testu :wink: 

`PpmToPbmConverter` jest  częścią innego projektu i tam jest przetestowany i o ile faktycznie jesteś w stanie zweryfikować jego działanie, lepiej tego nie robić i nie testować zewnętrznych zależności. Zamiast tego, lepiej w imię testowalności wstrzykiwać instancję `PpmToPbmConverter` do serwisu jako zależność i użyć mockito by ją zamockować. W teście wystarczy zweryfikować czy mock został poprawnie użyty (czy przekazano mu strumień jako parametr i czy zwracany output stream jest poprawnie obsługiwany.

W tym celu możesz np. zamockować to tak, by zwracany `OutputStream` był po prostu `ByteArrayOutputStream` z jakąś tablicą bajtów (choćby jednoelementową, to nieistotne) i potem asercją sprawdzić, czy stworzony `InputStream` zwracany z metody serwisowej zawiera tę tablicę. To nie powinno być zanadto kłopotliwe - AssertJ ma gotowe asercje na strumienie.

Pamiętaj: to test **jednostkowy** czyli powinien testować jak najmniejsze fragmenty aplikacji, idealnie pojedyncze metody, a jeśli wywoływane są inne, to je mockować.